pkg/controller/build/build_controller.go

```diff
@@ -252,6 +250,7 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 	b.Status.Reason = build.SucceedStatus
 
 	// Validate if the referenced secrets exist in the namespace
+	validationFailed = false
```

Comment:
@qu1queee But if we only declare it at the beginning, it will cause `validationFailed` alway is `true` sometimes. For example:
 First, Build validate `secret`, it cause `validationFailed` is equal `true`. Next, Build will validate `buildStrategy` and the buildStrategy exists. Then to be honest, Build should reset `validationFailed` to be `false` because reconcile is unnecessary. However, Build doesn't do this.

If we only want to set it at the beginning, then we must need to reset it at each validation function. Otherwise, we need to set it multiple times.